pkg/util/mempool/pool.go

```diff
@@ -54,8 +48,8 @@ func (s *slab) get(size int) ([]byte, error) {
 
 	waitStart := time.Now()
 	// wait for available buffer on channel
-	ptr := <-s.buffer
-	buf := unsafe.Slice((*byte)(ptr), s.size) //#nosec G103 -- Simple arena allocator implementation, does not appear to allow for any unsafe operations.
+	buf := <-s.buffer
+	buf = buf[:size]
```

Comment:
Already done at line 55 when the slice is returned

Reply:
Resolved by https://github.com/grafana/loki/pull/15428/commits/e2cd39434c720baf4886b522cfdb8ed6c38f3625